source/runtime.cpp

```diff
@@ -905,12 +911,24 @@ void reshade::runtime::save_config() const
 	config.set("GENERAL", "IntermediateCachePath", _effect_cache_path);
 
 	// Use ReShade DLL directory as base for relative preset paths (see 'resolve_preset_path')
-	std::filesystem::path relative_preset_path = _current_preset_path.lexically_proximate(g_reshade_base_path);
-	if (relative_preset_path.native().rfind(L"..", 0) != std::wstring::npos)
-		relative_preset_path = _current_preset_path; // Do not use relative path if preset is in a parent directory
-	if (relative_preset_path.is_relative()) // Prefix preset path with dot character to better indicate it being a relative path
-		relative_preset_path = L"." / relative_preset_path;
-	config.set("GENERAL", "PresetPath", relative_preset_path);
+	std::filesystem::path startup_preset_path = (g_reshade_base_path / _startup_preset_path).lexically_proximate(g_reshade_base_path);
+
+	if (startup_preset_path.native().rfind(L"..", 0) != std::wstring::npos)
+		startup_preset_path = _startup_preset_path; // Do not use relative path if preset is in a parent directory
+	if (startup_preset_path.is_relative()) // Prefix preset path with dot character to better indicate it being a relative path
+		startup_preset_path = L"." / startup_preset_path;
+
+	config.set("GENERAL", "StartupPresetPath", startup_preset_path);
+
+	// Use ReShade DLL directory as base for relative preset paths (see 'resolve_preset_path')
+	std::filesystem::path current_preset_path = (g_reshade_base_path / _current_preset_path).lexically_proximate(g_reshade_base_path);
```

Comment:
The `(g_reshade_base_path / _current_preset_path)` should not be necessary, as `_current_preset_path` is always guaranteed to be an absolute path (at load time it is made absolute via the call in `if (!resolve_preset_path(_current_preset_path, ec))`, and when changed in the overlay it is made absolute in `imgui::file_dialog()`). As such the previous code should already work.

Reply:
correct. This is the correct process I misunderstood.